crates/subspace-runtime/src/lib.rs

```diff
@@ -453,16 +453,19 @@ pub type Executive = frame_executive::Executive<
     AllPallets,
 >;
 
-fn extract_root_block(encoded_extrinsic: Vec<u8>) -> Option<RootBlock> {
-    if let Ok(extrinsic) = UncheckedExtrinsic::decode(&mut encoded_extrinsic.as_slice()) {
-        if let Call::Subspace(pallet_subspace::Call::store_root_block { root_block }) =
-            extrinsic.function
-        {
-            return Some(root_block);
+trait ContainsRootBlock {
+    fn root_block(&self) -> Option<RootBlock>;
+}
+
+impl ContainsRootBlock for UncheckedExtrinsic {
+    fn root_block(&self) -> Option<RootBlock> {
+        match self.function {
+            Call::Subspace(pallet_subspace::Call::store_root_block { root_block }) => {
+                Some(root_block)
+            }
+            _ => None,
         }
     }
```

Comment:
This is weird, do we really need to introduce a trait? Plain function is better, especially when it matches the same name from the other trait. The primary reason it is here in the first place is again to help with IDE support that doesn't really understand code in proc macros very well.

Reply:
>do we really need to introduce a trait?

Yes, because it's not allowed to implement an inherent method for a type that comes from another crate in Rust.

Reply:
But with plain function you don't need any of that :thinking:

Reply:
Okay, got you, makes sense :(.